File: CS/MVVMDemo.Wpf/Modules/Commands/DelegateCommandsViewModel.cs
using DevExpress.Mvvm;
using System.Windows;
using System.Windows.Input;

namespace MVVMDemo.Commands {
    public class DelegateCommandsViewModel : ViewModelBase {
        public ICommand SaveCommand { get; private set; }
        public ICommand OpenCommand { get; private set; }

        #region !
        public DelegateCommandsViewModel() {
            CanExecuteSaveCommand = true;

            SaveCommand = new DelegateCommand(
                () => MessageBox.Show("Action: Save"),
                () => CanExecuteSaveCommand);

            OpenCommand = new DelegateCommand<string>(
                fileName => MessageBox.Show(string.Format("Action: Open {0}", FileName)),
                fileName => !string.IsNullOrEmpty(fileName));
        } 
        #endregion

        public bool CanExecuteSaveCommand {
            get { return GetProperty(() => CanExecuteSaveCommand); }
            set { SetProperty(() => CanExecuteSaveCommand, value); }
        }
        public string FileName {
            get { return GetProperty(() => FileName); }
            set { SetProperty(() => FileName, value); }
        }
    }
}

File: VB/MVVMDemo.Wpf/Modules/Commands/DelegateCommandsViewModel.vb
Imports DevExpress.Mvvm
Imports System.Windows
Imports System.Windows.Input

Namespace MVVMDemo.Commands
    Public Class DelegateCommandsViewModel
        Inherits ViewModelBase

        Private privateSaveCommand As ICommand
        Public Property SaveCommand() As ICommand
            Get
                Return privateSaveCommand
            End Get
            Private Set(ByVal value As ICommand)
                privateSaveCommand = value
            End Set
        End Property
        Private privateOpenCommand As ICommand
        Public Property OpenCommand() As ICommand
            Get
                Return privateOpenCommand
            End Get
            Private Set(ByVal value As ICommand)
                privateOpenCommand = value
            End Set
        End Property

        #Region "!"
        Public Sub New()
            CanExecuteSaveCommand = True

            SaveCommand = New DelegateCommand(Function() MessageBox.Show("Action: Save"), Function() CanExecuteSaveCommand)

            OpenCommand = New DelegateCommand(Of String)(Function(fileName) MessageBox.Show(String.Format("Action: Open {0}", FileName)), Function(fileName) (Not String.IsNullOrEmpty(fileName)))
        End Sub
        #End Region

        Public Property CanExecuteSaveCommand() As Boolean
            Get
                Return GetProperty(Function() CanExecuteSaveCommand)
            End Get
            Set(ByVal value As Boolean)
                SetProperty(Function() CanExecuteSaveCommand, value)
            End Set
        End Property
        Public Property FileName() As String
            Get
                Return GetProperty(Function() FileName)
            End Get
            Set(ByVal value As String)
                SetProperty(Function() FileName, value)
            End Set
        End Property
    End Class
End Namespace
